src/rpc/server.cpp

```diff
@@ -209,11 +209,6 @@ std::string CRPCTable::help(std::string strCommand) const
             continue;
         if ((strCommand != "" || pcmd->category == "hidden") && strMethod != strCommand)
             continue;
-#ifdef ENABLE_WALLET
-        if (pcmd->reqWallet && !pwalletMain)
-            continue;
-#endif
-
```

Comment:
Without `EnsureWalletIsAvailable`, doesn't this mean that the wallet commands will be listed in the help, even when wallet is disabled?

Reply:
Not if the commands are not registered. `mapCommands` stores all the available commands. Wallet rpc commands are appended  on `walletRegisterRPCCommands()`, which is only called when the wallet `ENABLE_WALLET` compile time flag is true.